### PR TITLE
[Loan] add module for loan 

### PR DESCRIPTION
**Pull request by lalocesar2003**
## Description 

- Adding a Loan Module inside `package/hub-score`

## Summary of changes 

- [ ] add endpoints: 

1. loan/status :  approved/refused
2. loan/repaymentTime : repaymentTime
3. loan/repaid : repaid
4. loan/repaymentDetails : tranches details ([amounts and their corresponding dates])

- [ ] -e2e test for each endpoint

### CODE REVIEW

File: packages/hub-score/src/modules/loans/dto/update-loan.dto.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export class UpdateLoanDto {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/dto/update-loan.dto.ts]: is only for testting i removed this in the next commit

File: packages/hub-score/src/modules/loans/loan.entity.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity()
+export class Loan {
+  @PrimaryGeneratedColumn()
+  id: number;
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loan.entity.ts]: okey

File: packages/hub-score/src/modules/outputs/output.entity.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity()
+export class Output {
+  @PrimaryGeneratedColumn()
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/outputs/output.entity.ts:5]: removed

File: packages/hub-score/src/modules/outputs/dto/update-output.dto.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export class UpdateOutputDto {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/outputs/dto/update-output.dto.ts]: removed

File: packages/hub-score/src/modules/outputs/dto/create-output.dto.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export class CreateOutputDto {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/outputs/dto/create-output.dto.ts:1]: removed

File: packages/hub-score/src/modules/loans/loans.service.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Loan } from './loan.entity';
+import { CreateLoanDto} from '../loans/dto/create-loan.dto'
+import { UpdateLoanDto } from './dto/update-loan.dto';
+
+@Injectable()
+export class LoanService {
+  constructor(
+    @InjectRepository(Loan)
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.service.ts]: the @Inject() decorator generate conflict with the repository

File: packages/hub-score/README.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+
+
+## Description
+
+This project aims to extract the credit score of each wallet address entered,and ensure that it qualifies for a loan.
```
Comment (mrhouzlane) [inline packages/hub-score/README.md:5]: This is not a project - rewrite this. 
It is not extracting credit score neither - we are talking about stats for borrowers here based on their loans. Their is not credit score.

File: packages/hub-score/README.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+## Installation
+
+```bash
+$ yarn install
+```
+
+# Setting up environment variables
+
+you need create a env in the score-cripto-main project in packages
```
Comment (mrhouzlane) [inline packages/hub-score/README.md:16]: this is deprecated

File: packages/hub-score/src/app.controller.spec.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+      controllers: [AppController],
+      providers: [AppService],
+    }).compile();
+
+    appController = app.get<AppController>(AppController);
+  });
+
+  describe('root', () => {
+    it('should return "Hello World!"', () => {
+      expect(appController.getHello()).toBe('Hello World!');
```
Comment (mrhouzlane) [inline packages/hub-score/src/app.controller.spec.ts:19]: Hello world ?

File: packages/hub-score/src/app.controller.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { Controller, Get } from '@nestjs/common';
+import { AppService } from './app.service';
+
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  @Get()
+  getHello(): string {
+    return this.appService.getHello();
```
Comment (mrhouzlane) [inline packages/hub-score/src/app.controller.ts:10]: hello ?

File: packages/hub-score/src/app.service.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class AppService {
+  getHello(): string {
+    return 'Hello World!';
```
Comment (mrhouzlane) [inline packages/hub-score/src/app.service.ts:6]: hello ?

**Review by mrhouzlane — CHANGES_REQUESTED** [review]

File: packages/hub-score/src/modules/outputs/output.entity.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity()
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/outputs/output.entity.ts]: the output module was deleted

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import { Controller, Get, Post, Put, Delete, Body, Param } from '@nestjs/common';
+import { LoanService } from '../loans/loans.service';
+import { CreateLoanDto} from '../loans/dto/create-loan.dto'
+import { UpdateLoanDto } from './dto/update-loan.dto';
+
+@Controller('loan')
+export class LoansController {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: changed
image

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 14]
```diff
@@ -0,0 +1,46 @@
+import { Controller, Get, Post, Put, Delete, Body, Param } from '@nestjs/common';
+import { LoanService } from '../loans/loans.service';
+import { CreateLoanDto} from '../loans/dto/create-loan.dto'
+import { UpdateLoanDto } from './dto/update-loan.dto';
+
+@Controller('loan')
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: changed 
image

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 15]
```diff
@@ -0,0 +1,46 @@
+import { Controller, Get, Post, Put, Delete, Body, Param } from '@nestjs/common';
+import { LoanService } from '../loans/loans.service';
+import { CreateLoanDto} from '../loans/dto/create-loan.dto'
+import { UpdateLoanDto } from './dto/update-loan.dto';
+
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: changed 
image

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+@Controller('loan')
+export class LoansController {
+  constructor(private loanService: LoanService) {}
+
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: added in the last commit

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+  @Post()
+  create(@Body() createLoanDto: CreateLoanDto) {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: removed

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    return this.loanService.create(createLoanDto);
+  }
+
+  @Get()
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: i create two new routes for check if the status is approved or refused

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+  findAll() {
+    return this.loanService.findAll();
+  }
+
+  @Get(':id')
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: removed

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+  findOne(@Param('id') id: number) {
+    return this.loanService.findOne(id);
+  }
+
+  @Put(':id')
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: removed

File: packages/hub-score/src/modules/loans/loans.controller.ts [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+  update(@Param('id') id: number, @Body() updateLoanDto: UpdateLoanDto) {
+    return this.loanService.update(id, updateLoanDto);
+  }
+
+  @Delete(':id')
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.controller.ts]: removed in the last commit``

File: packages/hub-score/src/modules/loans/loans.module.ts [hunk 22]
```diff
@@ -0,0 +1,12 @@
+import { Module } from '@nestjs/common';
+import { LoanService } from '../loans/loans.service';
+import { LoansController } from './loans.controller';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Loan } from './loan.entity';
+
+@Module({
+  imports: [TypeOrmModule.forFeature([Loan])],
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.module.ts]: changed 
image

File: packages/hub-score/src/modules/loans/loans.module.ts [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
+  providers: [LoanService],
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.module.ts]: ``@Module({
  imports: [TypeOrmModule.forFeature([LoanEntity])],
  controllers: [LoanController],
  providers: [LoanService,LoanRepository],
  exports:[LoanService],
})`` changed

[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loans/loans.module.ts]: changed `@Module({`
  imports: [TypeOrmModule.forFeature([LoanEntity])],
  controllers: [LoanController],
  providers: [LoanService,LoanRepository],
  exports:[LoanService],
})``

File: packages/hub-score/src/modules/loan/loan.controller.ts [hunk 24]
```diff
@@ -0,0 +1,48 @@
+import { Controller, Get, Post, Put, Delete, Body, Param } from '@nestjs/common';
+import { LoanService } from './loan.service';
+
+import { ApiTags } from '@nestjs/swagger';
+import { LoanEntity } from './loan.entity';
+import { LoanRepaidDto, LoanRepaymentDetailsDto, LoanRepaymentTimeDto } from './loan.dto';
+
+@ApiTags('LOAN')
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loan/loan.controller.ts:8]: changed 
image

File: packages/hub-score/src/modules/loan/loan.controller.ts [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+@Controller('loan')
+export class LoansController {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loan/loan.controller.ts:10]: modified image

File: packages/hub-score/src/modules/loan/loans.service.spec.ts [hunk 26]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { LoansService } from './loan.service';
+
+describe('LoansService', () => {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loan/loans.service.spec.ts:4]: changed 
image

File: packages/hub-score/src/modules/outputs/dto/update-loan.dto.ts [hunk 27]
```diff
@@ -0,0 +1,4 @@
+export class UpdateOutputDto {
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/outputs/dto/update-loan.dto.ts]: removed

File: packages/hub-score/src/app.service.ts (hunk 11 above)
Comment (lalocesar2003) [reply packages/hub-score/src/app.service.ts:6]: the route was deleted

File: packages/hub-score/src/app.controller.ts (hunk 10 above)
Comment (lalocesar2003) [reply packages/hub-score/src/app.controller.ts:10]: the route was deleted

File: packages/hub-score/src/app.controller.spec.ts (hunk 9 above)
Comment (lalocesar2003) [reply packages/hub-score/src/app.controller.spec.ts:19]: the spec was deleted

File: packages/hub-score/README.md (hunk 7 above)
Comment (lalocesar2003) [reply packages/hub-score/README.md:5]: this is okey "This project aims to creates stadistic of each wallet address entered,for loans."

File: packages/hub-score/README.md (hunk 8 above)
Comment (lalocesar2003) [reply packages/hub-score/README.md:16]: i changed for hub-score 
image

File: packages/hub-score/src/modules/loan/loan.controller.ts [hunk 28]
```diff
@@ -0,0 +1,48 @@
...
+
+  @Get('repaid/:id')
+  async getRepaid(@Param('id') id: number): Promise<LoanRepaidDto> {
+    const repaid = await this.loanService.getRepaidById(id);
+    return { repaid };
+  }
+
+  @Get('repaymentDetails/:id')
+  async getRepaymentDetails(@Param('id') id: number): Promise<LoanRepaymentDetailsDto> {
+    const tranches = await this.loanService.getRepaymentDetailsById(id);
```
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loan/loan.controller.ts]: image
like this?

File: packages/hub-score/src/app.controller.ts (hunk 10 above)
Comment (lalocesar2003) [reply packages/hub-score/src/app.controller.ts]: is a default route for test the project

File: packages/hub-score/src/app.service.ts (hunk 11 above)
Comment (lalocesar2003) [reply packages/hub-score/src/app.service.ts]: i a default service method that generate when you create a nestjs project

File: packages/hub-score/src/modules/loan/loan.controller.ts (hunk 28 above)
[the remark this answers is not in the document]
Comment (lalocesar2003) [reply packages/hub-score/src/modules/loan/loan.controller.ts]: is this a good response ,if not can you give me a example

File: packages/hub-score/src/modules/loan/loan.controller.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/hub-score/src/modules/loan/loan.controller.ts]: you don't need to put a screenshot

File: packages/hub-score/src/modules/loan/loan.controller.ts (hunk 25 above)
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/hub-score/src/modules/loan/loan.controller.ts]: why are u sharing a screenshot ?

**Pull request closed without merging**